lib/licensed/sources/gradle.rb

```diff
@@ -30,55 +52,33 @@ def enabled?
       end
 
       def enumerate_dependencies
-        @dependencies ||= JSON.parse(package_metadata_command).map do |package|
+        packages.map do |package|
           name = "#{package["group"]}:#{package["name"]}"
           Dependency.new(
             name: name,
             version: package["version"],
-            path: File.join(@config.pwd, GRADLE_LICENSES_PATH, name),
+            path: package["url"],
             metadata: {
               "type"     => Gradle.type
             }
           )
         end
       end
 
-      def with_latest_licenses
-        download_and_cache_licenses do
-          yield
-        end
-      end
-
       private
 
-      def download_licenses_from_csv
+      def packages
+        metadata = JSON.parse(gradle_command("printDependencies"))
+        gradle_command("generateLicenseReport")
         path = File.join(@config.pwd, GRADLE_LICENSES_PATH)
```

Comment:
ah, I think I also missed removing the licenses path and CSV file after evaluating the CSV contents